src/iam/resources.py

```diff
@@ -252,3 +254,24 @@ def get(self):
             return User.query.filter(User.id == g.jwt_claims['usr']).one(), 200
         except NoResultFound:
             return f"No user with id {g.jwt_claims['usr']}", 404
+
+
+@doc(description="Register a user and authenticate in the local database")
+class UserRegisterResource(MethodResource):
+    @use_kwargs(UserRegisterSchema)
+    def post(self, first_name, last_name, email, password):
+        # Check if specified email already exists
+        user = User.query.filter_by(email=email).first()
```

Comment:
I would use `.exists()` instead of `.first()` to limit user's data exposure

Reply:
Thanks for a good point, decided to use scalar() instead (which will return user id or None), exists() is more expensive because it uses a subquery